### PR TITLE
Revenants now have cool, revenant-y spanclasses

### LINKED ISSUE #12704: Revenants can't hear deadchat and like to talk on deadchat
This is somewhat confusing and mildly meme-y.
Relatedly, revenants might not be able to hear other revenants(and if they can it's impossible for them to tell what revenant is talking)

### PR DESCRIPTION
**Pull request by ChangelingRain**
> Revenants now have numbers so they don't all appear the same.
> Revenant talk is now a non-deadchat span class, so it doesn't appear to be deadchat.
> Overload lights now turns lights purple for the lightning duration, like the spell icon.
> Fixes #12704
> Fixed a bug where the 'no key found' message in reforming would always trigger and the key of the revenant to reform would never appear.

SHOCK THE ERPERS
Spooky lights!

### CODE REVIEW

File: code/modules/mob/living/simple_animal/revenant/revenant.dm [hunk 1]
```diff
@@ -370,36 +378,41 @@
...
 	var/mob/living/simple_animal/revenant/R = new(get_turf(src))
 	if(client_to_revive)
 		for(var/mob/M in mob_list)
-			if(M.client == client_to_revive && M.stat == DEAD) //Only recreates the mob if the mob the client is in is dead
-				message_admins("[R.client] was a revenant and died. Re-making them into the new revenant formed by ectoplasm.")
+			if(M.stat == DEAD && M.client == client_to_revive) //Only recreates the mob if the mob the client is in is dead
+				message_admins("[M.client] was a revenant and died. Re-making them into the new revenant formed by ectoplasm.")
 				R.client = client_to_revive
 				key_of_revenant = client_to_revive.key
 				break
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/revenant/revenant.dm]: This chunk, 394 to 399, doesn't fucking work right please inform me how to fix this and or where Xhuis is so I can punch his face.
Specifically, even if the client is in a non-dead mob, it'll still show the admin message and fail to try and get other candidates.

Comment (Giacom) [inline code/modules/mob/living/simple_animal/revenant/revenant.dm]: I think the stat var is irrelevant for non living mobs.

Comment (Incoming5643) [inline code/modules/mob/living/simple_animal/revenant/revenant.dm]: dead_mob_list is a thing

**Comment by duncathan** [comment]
>purple light fixtures

Colored lighting when :^)

**Comment by Anonus** [comment]
WHEN YOU CODE IT, THAT'S WHEN

**Pull request merged**